sale_procurement_analytic/models/sale.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def _prepare_order_line_procurement(self, group_id=False):
+        self.ensure_one()
+        res = super(SaleOrderLine, self)._prepare_order_line_procurement(
+            group_id=group_id)
+        res['account_analytic_id'] = self.order_id.related_project_id.id
```

Comment:
Better to use `project_id` directly. By the way, what a bad name! It's the analytic account, not the project. Please put a comment indicating that the field contains actually the analytic account, not the project, so that anyone that sees this code doesn't wonder the same